scl/pgsql/pgsql.conf

```diff
@@ -35,37 +35,111 @@
+                "leader_pid",
+                "query_id",
+            )
+            delimiters(",")
+            dialect(escape-double-char)
+            flags(strip-whitespace)
+            prefix(`prefix`)
+            quote-pairs('""')
+        );
+        };
+        # Add types to values
+        rewrite {
+            set(int("${`prefix`pid}") value("`prefix`pid") internal(yes));
+            set(int("$(if ('${`prefix`session_line_num}' ne '') ${`prefix`session_line_num} '')") value("`prefix`session_line_num") internal(yes));
+            set(int("$(if ('${`prefix`transaction_id}' ne '') ${`prefix`transaction_id} '')") value("`prefix`transaction_id") internal(yes));
+            set(int("$(if ('${`prefix`internal_query_pos}' ne '') ${`prefix`internal_query_pos} '')") value("`prefix`internal_query_pos") internal(yes));
+            set(int("$(if ('${`prefix`query_pos}' ne '') ${`prefix`query_pos} '')") value("`prefix`query_pos") internal(yes));
+
+        };
+        # Extract values into built-in macros
+        parser {
+          date-parser(
+            internal(yes)
+            # Extract without the specified timezone as it seems to be not standard compilant.
+            format("%Y-%m-%d %H:%M:%S.%f")
+            template("$(substr ${`prefix`timestamp} 0 23)")
+            );
+        };
+         rewrite {
+            set(int("${`prefix`pid}") value("PID") internal(yes));
+            set(string("$(if ('${`prefix`connection_from}' ne '') ${`prefix`connection_from} $HOST_FROM)") value("`prefix`connection_from") internal(yes));
+            set(string("${`prefix`message}") value("MESSAGE") internal(yes));
+        };
+        # Map severity
+        if (match("DEBUG[1-5]" value('`prefix`severity') type(pcre))) {
+            rewrite {
+                set-severity("DEBUG");
+            };
+        }
+        elif (match("INFO" value('`prefix`severity') type(string))) {
+            rewrite {
+                set-severity("INFO");
+            };
+        }
+        elif (match("NOTICE" value('`prefix`severity') type(string))) {
+            rewrite {
+                set-severity("NOTICE");
+            };
+        }
+        elif (match("WARNING" value('`prefix`severity') type(string))) {
+            rewrite {
+                set-severity("NOTICE");
+            };
+        }
+        elif (match("ERROR" value('`prefix`severity') type(string))) {
+            rewrite {
+                set-severity("WARNING");
+            };
+        }
+        elif (match("LOG" value('`prefix`severity') type(string))) {
+            rewrite {
+                set-severity("INFO");
+            };
+        }
+        elif (match("FATAL" value('`prefix`severity') type(string))) {
+            rewrite {
+                set-severity("ERR");
+            };
+        }
+        elif (match("PANIC" value('`prefix`severity') type(string))) {
+            rewrite {
+                set-severity("CRIT");
+            };
```

Comment:
These mappings seem to be a bit odd, when we have the exact severity, I think we should use that (warning, error).

I would do it like this:
* WARNING -> WARNING
* ERROR -> ERROR
* FATAL -> CRIT
* PANIC -> CRIT

Reply:
They are, however it mirrors how they translate it to syslog format (see `Table 20.2. Message Severity Levels`):
https://www.postgresql.org/docs/current/runtime-config-logging.html